src/components/JobsPanelDataInputs/JobsPanelDataInputs.js

```diff
@@ -56,78 +63,77 @@ const JobsPanelDataInputs = ({
   ])
 
   useEffect(() => {
     if (
       inputsState.artifacts.length === 0 &&
-      inputsState.newInputProjectPathEntered
+      inputsState.inputProjectPathEntered
     ) {
-      fetchArtifacts({ project: inputsState.newInput.path.project }).then(
-        artifacts => {
-          const artifactsList = artifacts
-            .map(artifact => ({
-              label: artifact.link_iteration
-                ? artifact.link_iteration.db_key
-                : artifact.key ?? '',
-              id: artifact.link_iteration
-                ? artifact.link_iteration.db_key
-                : artifact.key ?? ''
-            }))
-            .filter(artifact => artifact.label !== '')
-
-          inputsDispatch({
-            type: inputsActions.SET_ARTIFACTS,
-            payload: artifactsList
-          })
-        }
-      )
+      fetchArtifacts({
+        project:
+          inputsState.newInput.path.project ||
+          inputsState.selectedDataInput.data.path.value.split('/')[0]
+      }).then(artifacts => {
+        const artifactsList = artifacts
+          .map(artifact => ({
+            label: artifact.link_iteration
+              ? artifact.link_iteration.db_key
+              : artifact.key ?? '',
+            id: artifact.link_iteration
+              ? artifact.link_iteration.db_key
+              : artifact.key ?? ''
+          }))
+          .filter(artifact => artifact.label !== '')
+
+        inputsDispatch({
+          type: inputsActions.SET_ARTIFACTS,
+          payload: artifactsList
+        })
+      })
     }
   }, [
     fetchArtifacts,
     inputsState.artifacts.length,
     inputsState.newInput.path.project,
-    inputsState.newInputProjectPathEntered
+    inputsState.inputProjectPathEntered,
+    inputsState.selectedDataInput.data.path.value
   ])
 
   useEffect(() => {
-    if (pathScheme === MLRUN_STORAGE_INPUT_PATH_SCHEME) {
-      let matches = []
-
-      if (inputsState.newInputProjectPathEntered) {
-        matches = inputsState.artifacts.filter(artifact =>
-          artifact.id.startsWith(inputsState.newInput.path.artifact)
-        )
-      } else if (
-        inputsState.newInput.path.project.length > 0 &&
-        inputsState.newInput.path.project !== match.params.projectName
-      ) {
-        matches = inputsState.projects.filter(project => {
-          return project.id.startsWith(inputsState.newInput.path.project)
-        })
-      } else if (inputsState.newInput.path.pathType.length === 0) {
-        matches = [...inputsState.artifacts]
-      } else {
-        matches = [...inputsState.projects]
-      }
-
-      inputsDispatch({
-        type: inputsActions.SET_COMBOBOX_MATCHES,
-        payload: matches
-      })
+    if (
+      inputsState.newInput.path.pathType === MLRUN_STORAGE_INPUT_PATH_SCHEME ||
+      inputsState.selectedDataInput.data.path.pathType ===
+        MLRUN_STORAGE_INPUT_PATH_SCHEME
+    ) {
+      generateComboboxMatchesList(
+        inputsDispatch,
```

Comment:
don't pass inputsDispatch, return matches from function and make dispatch call here in component